.github/workflows/pr-checks.yml

```diff
@@ -0,0 +1,49 @@
+name: Run PR checks
+run-name: Checks for PR#${{ github.event.pull_request.number }}
+
+on:
+  pull_request:
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  lint:
+    name: Linter check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: pnpm/action-setup@v2
+        with:
+          version: latest
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 20.x
+          cache: pnpm
+      - name: Install dependencies
+        run: pnpm install
+      - name: Run prettier
+        run: pnpm lint
+
+  build:
+    name: Build check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: pnpm/action-setup@v2
+        with:
+          version: latest
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 20.x
+          cache: pnpm
+      - name: Install dependencies
+        run: pnpm install
+      - name: Run test build
```

Comment:
merge the lint into the build job. Add it above here. Most build jobs include the lint in them

Reply:
Do you mean remove the lint job entirely and add it as a step to the build job, or just also lint in the build job?

Reply:
Let's merge the jobs into one job (run it as a step in the build job)
People should be running lint before they commit / push. (we could also add lint to pre-push hook, but don't need to do that now)

downside is that it will take slightly longer if you are iterating on build issues in the pipeline
upside is a whole lot less cpu time.

the upsides are almost guaranteed, but the likelihood that we hit an issue in build that can only be debugged in the pipeline is low.

Reply:
I'll get the commit ready to make this happen, but I wanted 1 chance to push back on this.

On the PR screen, GitHub only shows checks with job-level granularity, not step.
![screenshot](https://github.com/comcode-org/hackmud_wiki/assets/1775803/fc277f13-f29e-46fc-aa4c-27675bc93d97)

When all the checks are steps in a single job, you lose the ability to see at-a-glance which check(s) your code failed to pass, and would instead need to open the runner logs.

Moreover, since steps inside a single job depend on the step before them, you may have problems with multiple checks that would only reveal themselves one at a time (i.e. failing the linting step would not tell you about a build/security/etc. problem until you made another commit and addressed them each in order).

If we're just talking about those of us in ComCODE, then none of that is really an obstacle, and it's realistic that we'll make sure to run `prettier` and test a build before opening a PR. But I think the long game here is that this repository will be used by the game's community, who will surely have varying degrees of competency with `git` and GitHub.

If we keep them separate, it's easy to say "oh if your PR has a red x on the line that says 'Linter checks' just make sure to run this command." If we group them together, I think we risk discouraging contribution by making it more arcane. I guess it might come down to whether we expect the workflow for the average player to be fork/edit/PR or if the majority of it will be using the "Edit this Page" button on a wiki article to open GitHub's web editor.

The problem could be largely solved like you said with pre-push hooks (although I'm not sure how that works with GitHub's web editor).

That being said, everything you mentioned is absolutely right. I agree action-specific build bugs are unlikely and CPU time would certainly be lowered. My understanding though is that it's free for public repos. And running checks in parallel might slightly decrease the TTL of page edits for players.

If you still think it's better as one job, I'll have the refactor ready to push, just wanted to offer this as food for thought.

Reply:
Also if we do stay the course: would the additional `dependency-review-action` check discussed above and below go in the same single job as well, or just lint/build specifically?

Reply:
My assumption is that we would not have outside contributors to anything but wiki content, and that we would be and need to be very careful with any changes to code. (aka, very low likelihood of any contributors hitting js linter errors)

That said, you're right about the cost of the CPU on the public repo -- and we should be able to improve the performance of the actions by caching the `pnpm install` somehow (there's usually a way to create dependencies and shared artifacts between job steps)

So lets start with the more usable split jobs for now. And also add the dep review as its own action as well. So three total.